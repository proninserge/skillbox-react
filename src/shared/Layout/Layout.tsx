import useUserData from '@utils/react/hooks/useUserData';
import useToken from '@utils/react/hooks/useToken';
import { useDispatch } from 'react-redux';
import styles from './layout.less';
import { usePostsData } from '@utils/react/hooks/usePostsData';
import {ActionCreator} from '@store/actions';
interface ILayoutProps {
    children?: React.ReactNode;
};

export function Layout({children}: ILayoutProps) {
    const dispatch = useDispatch();

    const [data] = useUserData();
    const [token] = useToken();
    const [posts] = usePostsData();

    React.useEffect(() => {
        if(data.name) {
            dispatch(ActionCreator.updateUserData(data));
        }

        if(token !== 'undefined') {
            dispatch(ActionCreator.updateToken(token));
        }

        if(posts[0]) {
            dispatch(ActionCreator.updatePosts(posts));
        }
    }, [data, token, posts]);

        
    return (
        <div className={styles.layout}>
            {children}
        </div>
    );
}