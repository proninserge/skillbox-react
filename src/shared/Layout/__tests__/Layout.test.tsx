import {shallow, configure} from 'enzyme';
import {Layout} from '../Layout';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Layout', () => {
    const wrapper = shallow(<Layout children={[<div key="qwe" className="child-one"/>, <div key="rty" className="child-two"/>]} />);

    test('should render with children', () => {
        expect(wrapper).toBeDefined();
        expect(wrapper.find('div.layout').isEmptyRender()).toBeFalsy();
        expect(wrapper.find('div.child-one').isEmptyRender()).toBeFalsy();
        expect(wrapper.find('div.child-two').isEmptyRender()).toBeFalsy();
    });

    // test ('should render snapshot', () => {
    //     expect(wrapper).toMatchSnapshot();
    // });

});