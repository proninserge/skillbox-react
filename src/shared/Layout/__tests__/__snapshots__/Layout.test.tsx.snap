// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render snapshot 1`] = `
<div
  className="layout"
>
  <div
    className="child-one"
    key="qwe"
  />
  <div
    className="child-two"
    key="rty"
  />
</div>
`;
